### PR TITLE
parser: add autoid cache option

### PR DESCRIPTION
**Pull request by AilinKid**
### What problem does this PR solve? 
By now, the auto-increment cache in TiDB is quite big for inserting performance.

In some cases, some table is used as a tool to generate some id, the autoid cache will miss when tidb crash or restart,  which leading autoid increment quickly.

The related to pingcap/tidb#15409

That's why we support the auto-increment cache option.

### What is changed and how it works?
Add `auto_increment_cache` table option.
Make some test about it.

### Check List 

Tests 

 - Unit test
 - Integration test

Related changes

 - Need to update the documentation

### CODE REVIEW

**Comment by tiancaiamao** [comment]
Is there a TiDB PR for this change?

**Comment by AilinKid** [comment]
> Is there a TiDB PR for this change?

Yes，it's still work in process.

**Comment by tiancaiamao** [comment]
> > Is there a TiDB PR for this change?
> 
> Yes，it's still work in process.

After it's done, please link that PR with this one, so we can merge them together. @AilinKid

File: parser.y [hunk 1]
```diff
@@ -4884,6 +4885,7 @@ UnReservedKeyword:
 |	"ADVISE"
 |	"ASCII"
 |	"AUTO_INCREMENT"
+|	"AUTO_INCREMENT_CACHE"
```
Comment (bb7133) [inline parser.y]: Please rename it to `AUTO_ID_CACHE`, as you mentioned, it doesn't affect AutoIncrement column only.

**Comment by bb7133** [comment]
Please address the comment and fix CI.

**Comment by bb7133** [comment]
LGTM, but let's hold it DNM until the TiDB PR is ready to merge.

**Pull request merged**